pool/example_test.go

```diff
@@ -0,0 +1,153 @@
+package pool_test
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"errors"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	"github.com/nspcc-dev/neofs-sdk-go/container"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/pool"
+	"github.com/nspcc-dev/neofs-sdk-go/session"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+	"github.com/nspcc-dev/neofs-sdk-go/waiter"
+)
+
+// Create pool instance with 3 nodes connection.
+// This InitParameters will make pool use 192.168.130.71 node while it is healthy.
+// Otherwise, it will make the pool use 192.168.130.72 for 90% of requests and 192.168.130.73 for remaining 10%.
+func ExampleNewPool() {
+	// import "github.com/nspcc-dev/neofs-sdk-go/user"
+
+	var signer user.Signer
+	var prm pool.InitParameters
+	prm.SetSigner(signer)
+	prm.AddNode(pool.NewNodeParam(1, "192.168.130.71", 1))
+	prm.AddNode(pool.NewNodeParam(2, "192.168.130.72", 9))
+	prm.AddNode(pool.NewNodeParam(2, "192.168.130.73", 1))
+	// ...
+
+	p, err := pool.NewPool(prm)
+
+	_ = p
+	_ = err
+}
+
+func ExampleNew_easiestWay() {
+	// Signer generation, like example.
+	pk, _ := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	signer := user.NewAutoIDSignerRFC6979(*pk)
+
+	p, _ := pool.New(
+		pool.NewFlatNodeParams([]string{"grpc://localhost:8080", "grpcs://localhost:8081"}),
+		signer,
+		pool.DefaultOptions(),
+	)
+	_ = p
+}
+
+func ExampleNew_adjustingParameters() {
+	// Signer generation, like example.
+	pk, _ := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	signer := user.NewAutoIDSignerRFC6979(*pk)
+
+	opts := pool.DefaultOptions()
+	opts.SetErrorThreshold(10)
+
+	p, _ := pool.New(
+		pool.NewFlatNodeParams([]string{"grpc://localhost:8080", "grpcs://localhost:8081"}),
+		signer,
+		opts,
+	)
+	_ = p
+}
+
+func ExamplePool_ObjectPutInit_explicitAutoSessionDisabling() {
```

Comment:
Doesn't look like an example, too small for it.

Reply:
You can add some comments to `ObjectPutInit` call about expected operation results in various cases.